marin/transform/lingoly/to_dolma.py

```diff
@@ -0,0 +1,90 @@
+import json
+import os
+import tempfile
+import zipfile
+from dataclasses import dataclass
+
+import draccus
+import fsspec
+import ray
+
+from marin.utils import fsspec_mkdirs
+
+
+@dataclass
+class ConvertLingolyToDolmaConfig:
+    input_path: str
+    """The directory containing the zip file of the lingoly dataset."""
+    output_path: str
+    """The directory to save the lingoly dataset in dolma format."""
+
+
+@ray.remote
+def _convert_lingoly_to_dolma(config: ConvertLingolyToDolmaConfig) -> None:
+    """Convert the lingoly dataset to dolma format by concatenating the preamble, context, and questions."""
+    # Create a temporary directory to extract the zip file
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Handle zip file from GCS or other cloud storage
+        local_zip_path = os.path.join(temp_dir, "lingoly.zip")
+
+        # Download the zip file if it's in cloud storage
+        with fsspec.open(config.input_path, "rb") as remote_file:
+            with open(local_zip_path, "wb") as local_file:
+                local_file.write(remote_file.read())
```

Comment:
fsspec AbstractFileSystem has a `get_file` which is probably better to use `get_file`